backend/src/main/kotlin/blue/mild/covid/vaxx/service/IsinValidationService.kt

```diff
@@ -23,14 +24,43 @@ class IsinValidationService(
     override suspend fun validatePatient(
         firstName: String,
         lastName: String,
-        personalNumber: String
+        personalNumber: String?,
+        insuranceNumber: String?
     ): IsinValidationResultDto {
-        val result = runCatching {
-            isinService.getPatientByParameters(
-                firstName = firstName,
-                lastName = lastName,
-                personalNumber = personalNumber
-            )
+        return runCatching {
+            if (personalNumber != null) {
+                val result = isinService.getPatientByParameters(
+                    firstName = firstName,
+                    lastName = lastName,
+                    personalNumber = personalNumber
+                )
+                convertResult(result)
+            } else if (insuranceNumber != null){
+                val byInsuranceNumberResult = isinService.getForeignerByInsuranceNumber(
+                    insuranceNumber = insuranceNumber
+                )
+                val validationResult = convertResult(byInsuranceNumberResult)
+                if (validationResult.status == PatientValidationResult.PATIENT_FOUND) {
+                    validationResult
+                } else {
+                    logger.info {
+                        "Foreigner was not found in ISIN by insurance number ${insuranceNumber}. Trying to found the " +
+                        "patient by first name, last name and personal number "
+                    }
+                    val byPersonalNumberResult = isinService.getPatientByParameters(
+                        firstName = firstName,
+                        lastName = lastName,
+                        personalNumber = insuranceNumber
+                    )
+                    convertResult(byPersonalNumberResult)
+                }
+            } else {
+                logger.error {
+                    "Both personal and insurance numbers are not set for patient ${firstName} ${lastName}. " +
+                    "This should not happen. Skipping ISIN validation."
+                }
+                IsinValidationResultDto(status = PatientValidationResult.WAS_NOT_VERIFIED)
```

Comment:
no to teda nevim, tenhle pripad bych handloval pred tim `runCatching` a vyhodil na tvrdo nejakou exception, ze wtf,  protoze pokud to dobre chapu, tak tohle by prislo z FE v tomhle stavu a to teda nechceme

Reply:
protoze co se stane tady je to, ze se flow registrace bude pokracovat dal a dostane se az tesne pred databazi, kde by se to melo snad zariznout